graders/c/cgrader/cgrader.py

```diff
@@ -115,7 +113,7 @@ def compile_file(
         std_obj_files = []
         objs = []
         for std_c_file in c_file if isinstance(c_file, list) else [c_file]:
-            obj_file = re.sub("\.[^.]*$", "", std_c_file) + ".o"
+            obj_file = re.sub(r"\.[^.]*$", "", std_c_file) + ".o"
```

Comment:
Isn't using the `r` triggering a raw encoding of the string making the `\` added in the pattern string redundant?

Reply:
I think that the `\.` is supposed to be part of the pattern (not escaping out the period). The issue is that the linter was complaining about `\.` being an invalid escape sequence.

Reply:
Actually the raw encoding is necessary (and the original should have had a `\\`, my mistake). In this case the regular expression is looking for a literal `.`, not its interpretation as "any character".